### PR TITLE
[feat] OpenAI: add support for GPT4-Turbo preview

### PR DESCRIPTION
**Pull request by torrmal**
we should just pass through s…tuff, anyway, adding support for gpt4 1106

@paxcema 

## Description

Please include a summary of the change and the issue it solves. 

Fixes #issue_number

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by paxcema** [comment]
The tuples this PR modifies are structured this way because we have some handlers that depend on what is supported on the base OpenAI handler (i.e. this one). In particular, Langchain and Anyscale check these groups before running their own set of custom procedures.

File: mindsdb/integrations/handlers/openai_handler/constants.py [hunk 1]
```diff
@@ -17,4 +17,6 @@
                 tuple(f'code-search-{model}-text-001' for model in COMPLETION_LEGACY_BASE_MODELS) + \
                 tuple(f'code-search-{model}-code-001' for model in COMPLETION_LEGACY_BASE_MODELS)
 
-ALL_MODELS = list(set(CHAT_MODELS + COMPLETION_MODELS + COMPLETION_LEGACY_MODELS + EMBEDDING_MODELS))  # noqa
\ No newline at end of file
+IMAGE_MODELS = ('dall-e-3')
```
Comment (paxcema) [inline mindsdb/integrations/handlers/openai_handler/constants.py]: This should be a tuple:

```
IMAGE_MODELS = ('dall-e-3',)
```

Notice that pesky little comma at the end which implies the type is `tuple` 👀

**Review by paxcema — APPROVED** [review]

File: mindsdb/integrations/handlers/openai_handler/constants.py (hunk 1 above)
Comment (paxcema) [reply mindsdb/integrations/handlers/openai_handler/constants.py]: Fixed. Tried it out and everything seems to be working.

**Comment by paxcema** [comment]
Changing PR title for a more descriptive one.

**Comment by martyna-mindsdb** [comment]
Hey @paxcema 
We use the image mode to generate images with OpenAI - does it use the Dalle model?
```
CREATE MODEL mindsdb.dalle
PREDICT img_url
USING
   engine = 'openai',
   mode = 'image',
   prompt_template = '{{text}}, 8K | highly detailed realistic 3d oil painting style cyberpunk by MAD DOG JONES combined with Van Gogh  |  cinematic lighting | happy colors';
```
Shall we include this Dalle model in the `IMAGE_MODELS` constant, instead of having this image mode? So we could create it like this:
```
CREATE MODEL mindsdb.dalle
PREDICT img_url
USING
   engine = 'openai',
   model_name = 'dall-e-3',      -- or model_name = 'dall-e',
   prompt_template = '{{text}}, 8K | highly detailed realistic 3d oil painting style cyberpunk by MAD DOG JONES combined with Van Gogh  |  cinematic lighting | happy colors';
```

**Comment by paxcema** [comment]
@martyna-mindsdb it does, by default the model is `dalle-2`. Right now the new tuple added in this PR is not being used anywhere. We can add `dalle-2` to it, but the actual model cannot be changed yet for the image mode when actually calling the API.

**Comment by martyna-mindsdb** [comment]
@paxcema 
Ok - so is it possible now to use Dalle3 by passing `model_name = 'dall-e-3'` in `CREATE MODEL` (like above)?

**Comment by martyna-mindsdb** [comment]
image

**Comment by paxcema** [comment]
@martyna-mindsdb as mentioned above, the `IMAGE_MODELS` tuple this PR adds has no effect as of yet, so `model_name = 'dall-e-3'` should not have any impact, we can only use DallE 2 right now. It should be enough if you do `mode='image'`.

**Comment by martyna-mindsdb** [comment]
@paxcema 
Thanks for clarifying that.
So this PR does not enable using Dalle3 model within MindsDB.

**Pull request merged**